rclcpp_lifecycle/src/lifecycle_node_interface_impl.hpp

```diff
@@ -179,7 +181,10 @@ class LifecycleNode::LifecycleNodeInterfaceImpl
       throw std::runtime_error(
               "Can't get state. State machine is not initialized.");
     }
-    resp->success = change_state(req->transition.id);
+    rcl_lifecycle_ret_t error;
+    auto ret = change_state(req->transition.id, error);
+    (void) error;
+    resp->success = (ret == RCL_RET_OK) ? true : false;
```

Comment:
The expression in the parenthesis is already the boolean you want. No need to the ternary operator here.